src/App.jsx

```diff
@@ -14,70 +16,70 @@ export const goods = [
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const DEFAULT_GOOD = 'Jam';
+  const EMPTY_VALUE = '';
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const [selectedGood, setSelectedGood] = useState(DEFAULT_GOOD);
+  const isSelectedGood = good => good === selectedGood;
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood
+          ? `${selectedGood} is selected`
+          : 'No goods selected'
+        }
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+        {!!selectedGood && (
+          <button
+            onClick={() => setSelectedGood(EMPTY_VALUE)}
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+          />
+        )}
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              className={cn({
+                'has-background-success-light': isSelectedGood(good),
+              }) || null}
+              key={good}
             >
-              +
-            </button>
-          </td>
+              <td>
+                <button
+                  onClick={() => (isSelectedGood(good)
+                    ? setSelectedGood(EMPTY_VALUE)
+                    : setSelectedGood(good)
+                  )}
+                  className={cn('button', {
+                    'is-info': isSelectedGood(good),
+                  })}
+                  data-cy={isSelectedGood(good)
+                    ? 'RemoveButton'
+                    : 'AddButton'
+                  }
+                  type="button"
+                >
+                  {isSelectedGood(good)
+                    ? '-'
+                    : '+'
```

Comment:
It looks a bit complicated. I think it's better just to render two different buttons to improve code readability